sqlglot/generator.py

```diff
@@ -306,7 +306,11 @@ def column_sql(self, expression):
     def columndef_sql(self, expression):
         column = self.sql(expression, "this")
         kind = self.sql(expression, "kind")
-        not_null = " NOT NULL" if expression.args.get("not_null") else ""
+        not_null = (
```

Comment:
should we instead refactor ColumnDef to take constraints / options ?

Reply:
I was trying to work with what was there but agree that a refactor is probably in order. see if this feels better